artemis-cli/src/main/java/org/apache/activemq/artemis/cli/commands/tools/InitLogging.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.activemq.artemis.cli.commands.tools;
+
+import java.io.File;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.lang.invoke.MethodHandles;
+import java.util.Collections;
+
+import io.airlift.airline.Command;
+import io.airlift.airline.Option;
+import org.apache.activemq.artemis.cli.commands.ActionAbstract;
+import org.apache.activemq.artemis.cli.commands.ActionContext;
+import org.apache.activemq.artemis.cli.commands.Create;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Command(name = "init-logging", description = "Initialize a default log4j configuration")
+public class InitLogging extends ActionAbstract {
+
+   private static final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+   public static final String OLD_LOG_NAME = "logging.properties";
+
+   @Option(name = "--force", description = "Overwrite configuration at destination directory")
+   private boolean force;
+
+   @Override
+   public Object execute(ActionContext context) throws Exception {
+      super.execute(context);
+
+      initLogging(force, new File(getBrokerInstance()));
+
+      return null;
+   }
+
+
+   public static void initLogging(boolean force, File brokerInstance) throws Exception {
+      File etc = new File(brokerInstance, "etc");
+      File newLogging = new File(etc, Create.ETC_LOG4J2_PROPERTIES);
+
+      if (!force && newLogging.exists()) {
+         System.err.println("A configuration file " + newLogging.getAbsolutePath() + " already exists! Use the option -f if you want to re-initialize it");
+         return;
+      }
+
+      Create.write("etc/" + Create.ETC_LOG4J2_PROPERTIES, newLogging, Collections.emptyMap(), false, force, "utf-8");
+   }
+
+   public static void verifyOlderLogging(File fileInstance) throws Exception {
+      File etc = new File(fileInstance, "etc");
+      File newLogging = new File(etc, Create.ETC_LOG4J2_PROPERTIES);
+      File oldLogging = new File(etc, OLD_LOG_NAME);
+
+      if (oldLogging.exists() && !newLogging.exists()) {
+         logger.warn("Logging not initialized!");
```

Comment:
Using logging to warn about logging doesnt necessarily make sense. In particular it will probably do nothing if there is no config, as the default is Error, and if there is configuration via one of the many other valid routes this doesnt consider, it will then log this incorrectly.

Reply:
It is calling logger.warn, and System.out

Reply:
Instead of telling the user call init-logger to fix it.. I'm just fixing itself with a System.out.


I will check why it's calling the warning twice.. and where the Jboss Logging is leaking into this.

Reply:
Yes, and I'm saying I really dont think we should try to fix it ourselves, as I did time you tried, because we cant really fix it properly ourselves, and we cant be sure of the effect doing it will have. Even with what you did it still has no effect until the next startup even if they dont have alternative valid config and it isnt interfering with other config.

The JBossLogging stuff comes in via the scripts hardcoding it, scripts that also wont have been updated if you dont follow the update instructions.